src/client/terminals/codeExecution/codeExecutionManager.ts

```diff
@@ -96,6 +97,7 @@ export class CodeExecutionManager implements ICodeExecutionManager {
     private async executeSelection(executionService: ICodeExecutionService): Promise<void> {
         const activeEditor = this.documentManager.activeTextEditor;
         if (!activeEditor) {
+            vscode.window.showErrorMessage('Open an active editor before executing code');
```

Comment:
Also please use existing shell functionality instead of importing `vscode` directly.
```suggestion
           const appShell = this.serviceContainer.get<IApplicationShell>(IApplicationShell);
           appShell.showErrorMessage('Open an active editor before executing code');
```